lib/kubernetes-deploy/runner.rb

```diff
@@ -216,17 +254,38 @@ def split_templates(filename)
         yield f
       end
     rescue Psych::SyntaxError => e
-      @logger.error(rendered_content)
-      raise FatalDeploymentError, "Template #{filename} cannot be parsed: #{e.message}"
+      debug_msg = <<-INFO.strip_heredoc
+      Error message: #{e}
+
+      Template content:
+      ---
+      INFO
+      debug_msg += rendered_content
+      @logger.summary.add_paragraph(debug_msg)
+      raise FatalDeploymentError, "Template '#{filename}' cannot be parsed"
     end
 
-    def report_final_status(resources)
-      if resources.all?(&:deploy_succeeded?)
-        @logger.info("Deploy succeeded!")
+    def record_apply_failure(err)
+      file_name, file_content = find_bad_file_from_kubectl_output(err)
+      if file_name
+        debug_msg = <<-HELPFUL_MESSAGE.strip_heredoc
+        This usually means your template named '#{file_name}' is invalid.
```

Comment:
Nitpicks but it's easier to read if the message is indented 2 spaces in the heredoc block
```ruby
<<-MSG
  hello
MSG

```